packages/cli/src/modules/migrate/commands/versions/migrate.ts

```diff
@@ -94,20 +96,32 @@ export async function migrateMovedPackages(options?: {
         }
 
         const movedPackageName = packageInfo.backstage?.moved;
+        if (!movedPackageName) {
+          continue;
+        }
 
-        if (movedPackageName) {
-          movedPackages.set(depName, movedPackageName);
-          console.log(
-            chalk.yellow(
-              `Found a moved package ${depName}@${depVersion} -> ${movedPackageName} in ${pkgName} (${depType})`,
-            ),
-          );
+        movedPackages.set(depName, movedPackageName);
+        console.log(
+          chalk.yellow(
+            `Found a moved package ${depName}@${depVersion} -> ${movedPackageName} in ${pkgName} (${depType})`,
+          ),
+        );
+
+        didPackageChange = true;
+        didAnythingChange = true;
 
-          didPackageChange = true;
-          didAnythingChange = true;
+        delete depsObj[depName];
 
-          depsObj[movedPackageName] = depsObj[depName];
-          delete depsObj[depName];
+        if (depVersion.startsWith('backstage:')) {
+          const info = await fetchPackageInfo(movedPackageName);
+          depsObj[movedPackageName] = `^${info['dist-tags'].latest}`;
+        } else if (
+          hasYarnPlugin &&
+          movedPackageName.startsWith('@backstage/')
+        ) {
+          depsObj[movedPackageName] = 'backstage:^';
+        } else {
+          depsObj[movedPackageName] = depVersion;
         }
```

Comment:
I'm still thinking the logic here is wrong, or am I misunderstanding the state?

Feeling it would make sense to first check if we're using the yarn plugin and then target package is a `@backstage/*` package, then we can use `'backstage:^'`. If that is not the case we know for sure we can't use the Backstage protocol, so if the current version range uses the protocol, then look up the latest version, and otherwise keep using the existing range.

Reply:
@rugvip means you would ignore the "moved package" part and only focus on the package itself, right?